File: src/Shared/Logistics.Shared.Models/Trip/CreateTripLoadCommand.cs
using Logistics.Domain.Primitives.Enums;
using Logistics.Domain.Primitives.ValueObjects;

namespace Logistics.Shared.Models;

public record CreateTripLoadCommand
{
    public string? TempId { get; set; } // Temporary ID for mapping optimized stops

    public required string Name { get; set; }

    public required Address OriginAddress { get; set; }
    public required GeoPoint OriginLocation { get; set; }

    public required Address DestinationAddress { get; set; }
    public required GeoPoint DestinationLocation { get; set; }

    public decimal DeliveryCost { get; set; }
    public double Distance { get; set; }
    public LoadType Type { get; set; }
    public required Guid AssignedDispatcherId { get; set; }
    public required Guid CustomerId { get; set; }
}

File: src/Client/Logistics.OfficeApp/src/app/core/api/models/trip/create-trip-load.command.ts
import { AddressDto } from "../address.dto";
import { GeoPointDto } from "../geo-point.dto";
import { LoadType } from "../load";

export interface CreateTripLoadCommand {
  tempId?: string; // Temporary ID for mapping optimized stops
  name: string;
  originAddress: AddressDto;
  originLocation: GeoPointDto;
  destinationAddress: AddressDto;
  destinationLocation: GeoPointDto;
  deliveryCost: number;
  distance: number;
  type: LoadType;
  assignedDispatcherId: string;
  customerId?: string;
}
